resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -323,6 +323,15 @@ private[spark] object Config extends Logging {
       .stringConf
       .createOptional
 
+  val KUBERNETES_ALLOCATION_PODSALLOCATOR =
+    ConfigBuilder("spark.kubernetes.allocation.podsallocator")
+      .doc("Allocator to use for pods. Possible values are direct (the default) and statefulset." +
+        "Future version may add Job or replicaset.")
+      .version("3.3.0")
+      .stringConf
+      .checkValues(Set("direct", "statefulset"))
```

Comment:
Instead of this, shall we make this more extensible by using a class name here?

Reply:
I was going back and forth on that, do you think it's worthwhile to expose this as a developer API or just within Spark?

Reply:
+1 for developer API so as to enable the usage of other options (like replicaset or Job) as applicable to other developers.

Reply:
Ok works for me :)